[ERROR] Not authorized or project not found. Please check the 'SONAR_TOKEN' environment variable, the 'sonar.projectKey' and 'sonar.organization' properties, or contact the project administrator to verify the token's permissions.

While reviewing the following #12551 I noticed the following error:

<img width="1215" height="566" alt="Image" src="https://github.com/user-attachments/assets/7a84a5d6-1245-4068-9185-994491f8e1b0" />

At https://github.com/IQSS/dataverse/actions/runs/30493484094/job/90717789386?pr=12551 I'm seeing the following error:

`[ERROR] Not authorized or project not found. Please check the 'SONAR_TOKEN' environment variable, the 'sonar.projectKey' and 'sonar.organization' properties, or contact the project administrator to verify the token's permissions.`

Here's a screenshot:

<img width="1740" height="1275" alt="Image" src="https://github.com/user-attachments/assets/1a47bf13-f926-49c1-a7c0-92aeb249af44" />

This certainly results from our recent addition of a Sonar check:

- #12550 

I'm assuming the failure above happened because the pull request came from a fork.

How should we handle this? In this case @qqmyers is a member of the core dev team, so should we ask him to set up the credentials in his fork?

For random contributors, should we update the pull request template and dev guide to set expectations that this error is expected?

--- a/.github/workflows/maven_unit_test.yml
+++ b/.github/workflows/maven_unit_test.yml
@@ -159,15 +159,34 @@ jobs:
             continue-on-error: true
             run: >
                 mvn -B
-                -DrepoToken=${COVERALLS_SECRET} -DpullRequest=${{ github.event.number }}
+                -DrepoToken="${COVERALLS_SECRET}" -DpullRequest=${{ github.event.number }}
                 jacoco:report coveralls:report
+
+  # Note: The SONAR_TOKEN secret is required to push analysis results to Sonarcloud.
+  # We check for them here and subsequent jobs can rely on this to decide if they shall run.
+    check-secrets:
+      needs: integration-test
+      name: Check for Secrets Availability
+      runs-on: ubuntu-latest
+      outputs:
+        available: ${{ steps.secret-check.outputs.available }}
+      steps:
+        - id: secret-check
+          # perform secret check & put boolean result as an output
+          shell: bash
+          run: |
+            if [ "${{ secrets.SONAR_TOKEN }}" != '' ]; then
+                echo "available=true" >> "$GITHUB_OUTPUT";
+            else
+                echo "available=false" >> "$GITHUB_OUTPUT";
+            fi
 
     sonarcloud:
       runs-on: ubuntu-latest
-      needs: integration-test
+      needs: check-secrets
       name: SonarQube Analysis and Coverage
       # Only run in upstream repo - avoid unnecessary runs in forks (they don't have access to SONAR_TOKEN)
-      if: ${{ github.repository_owner == 'IQSS' }}
+      if: needs.check-secrets.outputs.available == 'true'
       steps:
         # TODO: As part of #10618 change to setup-maven custom action
         # Basic setup chores

--- a/.github/workflows/maven_unit_test.yml
+++ b/.github/workflows/maven_unit_test.yml
@@ -166,6 +166,8 @@ jobs:
       runs-on: ubuntu-latest
       needs: integration-test
       name: SonarQube Analysis and Coverage
+      # Only run in upstream repo - avoid unnecessary runs in forks (they don't have access to SONAR_TOKEN)
+      if: ${{ github.repository_owner == 'IQSS' }}
       steps:
         # TODO: As part of #10618 change to setup-maven custom action
         # Basic setup chores

--- a/pom.xml
+++ b/pom.xml
@@ -143,10 +143,10 @@
         </dependency>
         
         <!-- TECHNICAL DEBT
-             For a very long time, in a lots of places people used org.apache.commons.httpclient to access
-             HTTP resources, wrap responses in exceptions etc.
+             For a very long time, in a lot of places people used org.apache.commons.httpclient to access
+             HTTP resources, wrap responses in exceptions, etc.
              
-             This library had been used via an transitive dependency of sword2-server. During its recent updates,
+             This library had been used via a transitive dependency of sword2-server. During its recent updates,
              the dependency was dropped, went missing, eventually breaking our code. Re-adding this 2007 goodie here
              as a direct dependency.
              

